browser/src/plugin.ts

```diff
@@ -1,15 +1,61 @@
 import Allocator from './allocator';
 import { PluginConfig } from './manifest';
 
-class ExtismPluginCall {
+export default class ExtismPlugin {
+  moduleData: ArrayBuffer;
+  allocator: Allocator;
+  config?: PluginConfig;
+  vars: Record<string, Uint8Array>;
   input: Uint8Array;
   output: Uint8Array;
-  allocator: Allocator;
+  module?: WebAssembly.WebAssemblyInstantiatedSource;
+
+  constructor(moduleData: ArrayBuffer, config?: PluginConfig) {
+    this.moduleData = moduleData;
+    this.allocator = new Allocator(1024 * 1024);
+    this.config = config;
+    this.vars = {};
+    this.input = new Uint8Array();
+    this.output = new Uint8Array();
+  }
+
+  async getExportedFunctions() {
+    const module = await this._instantiateModule();
+    return Object.keys(module.instance.exports).filter((f) => !f.startsWith('__') && f !== 'memory');
```

Comment:
curious about the `__` prefix filter, do we really want to exclude these? I get it for the use-case in the Playground, but do we need it here? 

maybe it would be better to add a single getter for the instance stuff (imports/exports/etc).

Reply:
Yeah agreed. If this is copied from me, I used this to make the dropdown for the playground.

Reply:
It might make more sense to just give them all the exports

Reply:
I will update it - should we still filter out the `memory` export?

Reply:
personally, i'd go the route of adding getters for the things on `module.instance` we want to make available. not explicitly supporting a way to list the functions here. let a user of this library make that decision. 

so like on an instantiated `Plugin`, we could add class methods to `getImports()` and `getExports()`, which then could be used in whatever way a caller wants?